### ISSUE TITLE
issue with preact-compat and glamor since 3.12

### ISSUE THREAD

**Issue by conorhastings** [developit/preact-compat#330]
Hi! I'm having a super weird issue  with preact-compat and glamor. I've narrowed it down to this addition -- https://github.com/developit/preact-compat/blob/master/src/index.js#L92 , if i comment that out the issue goes away. 

Basically it seems like somehow glamor is getting a wrong function argument for the second instance of a component (like i said super weird). It only happens if the component has at least one child that is a non string.

I was able to make a relatively small reproducable test case -- https://esnextb.in/?gist=1911224ff4bc0b93a205b10c60b4be83, let me know if you have any ideas thanks!

**Comment by conorhastings** [comment]
Another interesting thing i've found is if the div in <Thing /> has any other properties in addition to the spread it works fine

**Comment by developit** [comment]
Ahh - sounds like we need to clone `attributes` before setting `.children`! Good catch - your explanation of the second instance thing helped narrow that down.

**Comment by conorhastings** [comment]
If the tree rerenders, all instances will become broken. So it seems like the only one that renders correctly, is the first instnce on the first render.

**Comment by conorhastings** [comment]
@developit ahh nice, I'd be happy to make a PR if you could point me in the rightish direction.

**Comment by developit** [comment]
Totally! I think we need to change [these lines](https://github.com/developit/preact-compat/blob/master/src/index.js#L74-L77) to clone attributes.

Basically swap this:

```js
let tag = vnode.nodeName,
	attrs = vnode.attributes;

if (!attrs) attrs = vnode.attributes = {};
```

... out for this:

```js
let tag = vnode.nodeName,
	attrs = vnode.attributes = extend({}, vnode.attributes);
```

**Comment by developit** [comment]
Looks like doing that fixes the issue: https://esnextb.in/?gist=fafa6fad1f8c6b3a9b44a622c1ee9a8b

**Comment by developit** [comment]
Released as [3.14.2](https://github.com/developit/preact-compat/releases/tag/3.14.2) 💃
